include/revng/Pipeline/Step.h

```diff
@@ -62,6 +68,24 @@ class Step {
   const ContainerSet &containers() const { return Containers; }
   ContainerSet &containers() { return Containers; }
 
+  void setArtifacts(const Kind *ArtifactsKind, std::string ContainerName) {
+    this->Artifacts.Kind = ArtifactsKind;
+    this->Artifacts.Container = ContainerName;
+  }
+
+  const Kind *getArtifactsKind() { return this->Artifacts.Kind; }
+
+  ContainerSet::value_type *getArtifactsContainer() {
```

Comment:
We probably prefer asserting that the artifact container exists and return a reference instead of trying to see if it exists and then returning nullptr if it does not. The reason being that it detects earlier missuses of the method.